Adding a "global" not_intent parameter for slot mappings (#7931)
RasaHQ/rasa: docs/docs/migration-guide.mdx

This page contains information about changes between major versions and
how you can migrate from one version to another.

## Rasa 2.5 to 2.6

### Forms

#### New `ignored_intents` parameter in Forms

We added a new parameter under Forms with the name `ignored_intents`. This parameter

> can be used in case the user never wants to fill any slots of a form with the specified

Reviewer: can we please show them an example which illustrates what they have to change exactly (like a "before" and "after" version of the domain)?

This page contains information about changes between major versions and
how you can migrate from one version to another.

## Rasa 2.5 to 2.6

### Forms

#### New `ignored_intents` parameter in Forms

We added a new parameter under Forms with the name `ignored_intents`. This parameter
can be used in case the user never wants to fill any slots of a form with the specified

> intent, eg. chitchat. Please see the [Forms documentation](forms.mdx) for examples and

Reviewer 2: ```suggestion
intent, e.g. chitchat. Please see the [Forms documentation](forms.mdx) for examples and
```

more information on how to use it in your `domain.yml` file.

> So before, if a user did not want to fill any slots of a form with a specified intent

Reviewer 2: ```suggestion
Before, if a user did not want to fill any slots of a form with a specified intent
```

> we would have to define it (with the `not_intent` parameter) in every slot
> as shown in the following example :

Reviewer 2: ```suggestion
they would have to define it under the `not_intent` parameter for every slot mapping
as shown in the following example :
```

```yaml
forms:
  restaurant_form:

>     required_slots:

Author: Yes, I thought about this but I added the `required_slots` in all examples, tests, docs, everywhere so I thought it would be weird to only have this place that `required_slots` keyword is missing. But ok, I removed it.

      cuisine:
      - type: from_entity
        entity: cuisine
        not_intent: chitchat
      num_people:
      - type: from_entity
        entity: number
        intent: [inform, request_restaurant]
        not_intent: chitchat
      feedback:
      - type: from_entity
        entity: feedback
        not_intent: chitchat
```

By introducing the `ignored_intents` parameter, we now only need to define it
in one place and it will affect all the slots of the form :

> ```yaml

Author: That is awesome! I did not know we could do that!

correct spaCy model. We've also added a warning to the [SpaCyNLP](components.mdx#spacynlp)
docs that explains the fallback behavior.

> ### Forms

Reviewer: that needs to go in a different section now, doesn't it? 2.6 to 2.5

Author: Moved it! The description is ok?

#### New `global_not_intent` parameter in Forms

We added a new parameter under Forms with the name `global_not_intent`. This parameter
can be used in case the user never wants to fill any slots of a form with the specified

> intent, eg. chitchat. Please see https://rasa.com/docs/rasa/forms for examples and more

Reviewer: please use relative links (ideally also link to the specific section in the forms docs page)
```suggestion
intent, eg. chitchat. Please see the [Forms documentation](forms.mdx) for examples and more
```

> intent, eg. chitchat. Please see https://rasa.com/docs/rasa/forms for examples and more

Reviewer: I'd vote for showing an example of the old format and then show how to transition that to the new format

> #### New `ignored_intents` parameter in Forms
> 
> We added a new parameter under Forms with the name `ignored_intents`. This parameter
> can be used in case the user never wants to fill any slots of a form with the specified
> intent, eg. chitchat. Please see the [Forms documentation](forms.mdx) for examples and more
> information on how to use it in your `domain.yml` file.
> 

Reviewer 2: ```suggestion
#### New `ignored_intents` parameter in Forms

There is a new parameter under Forms called `ignored_intents`. This parameter
can be used to prevent any required slots in a form from being filled with the specified
intent or intents. Please see the [Forms documentation](forms.mdx) for examples and more
information on how to use it in your `domain.yml` file.

```
